cmd/tegola/cmd/root.go

```diff
@@ -152,3 +156,124 @@ func initConfig(configFile string, cacheRequired bool, logLevel string, logger s
 	atlas.SetObservability(observer)
 	return nil
 }
+
+// initProviders translate provider config from a TOML file into usable Provider objects.
+func initProviders(providersConfig []env.Dict, maps []provider.Map) (map[string]provider.TilerUnion, error) {
+	// first convert []env.Map -> []dict.Dicter
+	provArr := make([]dict.Dicter, len(providersConfig))
+	for i := range provArr {
+		provArr[i] = providersConfig[i]
+	}
+
+	providers, err := register.Providers(provArr, conf.Maps)
+	if err != nil {
+		return nil, fmt.Errorf("could not register providers: %v", err)
+	}
+
+	return providers, nil
+}
+
+// initMaps registers maps with Atlas to be ready for service.
+func initMaps(maps []provider.Map, providers map[string]provider.TilerUnion) error {
+	if err := register.Maps(nil, maps, providers); err != nil {
+		return fmt.Errorf("could not register maps: %v", err)
+	}
+
+	return nil
+}
+
+// initAppConfigSource sets up an additional configuration source for "apps" (groups of providers and maps) to be loaded and unloaded on-the-fly.
+func initAppConfigSource(conf config.Config) error {
+	// Get the config source type. If none, return.
+	val, err := conf.AppConfigSource.String("type", nil)
+	if err != nil || val == "" {
+		return nil
+	}
+
+	// Initialize the source.
+	ctx := context.Background() // Not doing anything with context now, but could use it for stopping this goroutine.
```

Comment:
This function should take in a context. Have the calling function pass in context.Background if needed.

Reply:
👍